iceoryx_posh/include/iceoryx_posh/internal/popo/ports/server_port_user.hpp

```diff
@@ -69,10 +70,14 @@ class ServerPortUser : public BasePort
 
     /// @brief Allocate a response, the ownerhip of the SharedChunk remains in the ServerPortUser for being able to
     /// cleanup if the user process disappears
+    /// @param[in] requestHeader, the request header for the corresponding response
     /// @param[in] userPayloadSize, size of the user user-paylaod without additional headers
+    /// @param[in] userPayloadAlignment, alignment of the user user-paylaod without additional headers
     /// @return on success pointer to a ChunkHeader which can be used to access the chunk-header, user-header and
     /// user-payload fields, error if not
-    cxx::expected<ResponseHeader*, AllocationError> allocateResponse(const uint32_t userPayloadSize) noexcept;
+    cxx::expected<ResponseHeader*, AllocationError> allocateResponse(const RequestHeader* const requestHeader,
```

Comment:
Same here, the user has to perform a `nullptr` when success is returned. Could we add here a `std::reference_wrapper<ResponseHeader>` as well?